Lib/test/libregrtest/main.py

```diff
@@ -534,7 +536,13 @@ def _main(self, tests, kwargs):
             self.rerun_failed_tests()
 
         self.finalize()
-        sys.exit(len(self.bad) > 0 or self.interrupted)
+        if self.bad:
+            sys.exit(2)
+        if self.interrupted:
+            sys.exit(3)
+        if self.ns.fail_env_changed and self.environment_changed:
```

Comment:
"if len(self.bad) > 0" and "if self.bad" are equivalent, since self.bad is a list. Can you please elaborate? I don't understand your suggestion.